core/src/main/kotlin/org/evomaster/core/search/gene/collection/MapGene.kt

```diff
@@ -168,12 +126,16 @@ class MapGene<K, V>(
 
         return "{" +
                 elements.filter { f ->
-                    isPrintable(f)
-                }.map {f->
+                    isPrintable(f) && (f.first !is OptionalGene || f.first.isActive) && (f.second !is OptionalGene || f.second.isActive)
```

Comment:
do not check wrapper genes like `OptionalGene` directly. it is (should be) safer to use the method `getWrappedGene(klass)`